lib/core/src/system/web/dom/html/css3d_renderer.rs

```diff
@@ -0,0 +1,269 @@
+//! This module contains the Css3dRenderer, a struct used to render CSS3D elements.
+
+use crate::prelude::*;
+
+use crate::display::object::DisplayObjectData;
+use crate::display::camera::Camera2d;
+use crate::display::camera::camera2d::Projection;
+use crate::system::web::dom::html::Css3dObject;
+use crate::system::gpu::data::JsBufferView;
+use crate::system::web::Result;
+use crate::system::web::create_element;
+use crate::system::web::dyn_into;
+use crate::system::web::NodeInserter;
+use crate::system::web::StyleSetter;
+use crate::system::web::dom::DomContainer;
+use crate::system::web::dom::ResizeCallback;
+use super::eps;
+
+use nalgebra::Vector2;
+use nalgebra::Matrix4;
+use wasm_bindgen::prelude::wasm_bindgen;
+use wasm_bindgen::JsValue;
+use web_sys::HtmlElement;
+use basegl_system_web::get_element_by_id;
+
+
+// ===================
+// === Js Bindings ===
+// ===================
+
+mod js {
+    use super::*;
+    #[wasm_bindgen(module = "/src/system/web/dom/html/snippets.js")]
+    extern "C" {
+        #[allow(unsafe_code)]
+        pub fn setup_perspective(dom: &JsValue, znear: &JsValue);
+
+        #[allow(unsafe_code)]
+        pub fn setup_camera_orthographic(dom:&JsValue, matrix_array:&JsValue);
+
+        #[allow(unsafe_code)]
+        pub fn setup_camera_perspective
+        ( dom          : &JsValue
+        , near         : &JsValue
+        , matrix_array : &JsValue
+        );
+    }
+}
+
+/// Inverts Matrix Y coordinates.
+/// It's equivalent to scaling by (1.0, -1.0, 1.0).
+pub fn invert_y(mut m: Matrix4<f32>) -> Matrix4<f32> {
+    // Negating the second column to invert Y.
+    m.row_part_mut(1, 4).iter_mut().for_each(|a| *a = -*a);
+    m
+}
+
+#[allow(unsafe_code)]
+fn setup_camera_perspective
+(dom:&JsValue, near:f32, matrix:&Matrix4<f32>) { // Views to WASM memory are only valid as long the backing buffer isn't
+    // resized. Check documentation of IntoFloat32ArrayView trait for more
+    // details.
+    unsafe {
+        let matrix_array = matrix.js_buffer_view();
+        js::setup_camera_perspective(
+            &dom,
+            &near.into(),
+            &matrix_array
+        )
+    }
+}
+
+#[allow(unsafe_code)]
+fn setup_camera_orthographic(dom:&JsValue, matrix:&Matrix4<f32>) {
+    // Views to WASM memory are only valid as long the backing buffer isn't
+    // resized. Check documentation of IntoFloat32ArrayView trait for more
+    // details.
+    unsafe {
+        let matrix_array = matrix.js_buffer_view();
+        js::setup_camera_orthographic(&dom, &matrix_array)
+    }
+}
+
+
+
+// =========================
+// === Css3dRendererData ===
+// =========================
+
+#[derive(Debug)]
+pub struct Css3dRendererData {
```

Comment:
There is a lot of missing docs here. Please check the parent modules to see where missing_docs were disabled.

Reply:
@wdanilo this struct isn't really public. Only specifics structs are re-exported in `html.rs`:

```rust
mod css3d_system;
mod css3d_object;
mod css3d_renderer;

pub use css3d_object::Css3dObject;
pub use css3d_object::Css3dOrder;
pub use css3d_renderer::Css3dRenderer;
pub use css3d_system::Css3dSystem;
```